script.js

```diff
@@ -1,4 +1,294 @@
+  if (player1Score > player2Score) {
+    myOutputValue = `Congratz ${player1} for winning!!! <br><br>
+    ${player1} BEATS ${player2} with a score of ${player1Score} against ${player2Score} <br><br>
+    Click enter to start another new game`;
+    return myOutputValue;
+  } else if (player2Score > player1Score) {
+    myOutputValue = `Congratz ${player2} for winning!!! <br><br>
+     ${player2} BEATS ${player1} with a score of ${player2Score} against ${player1Score}<br><br>
+    Click enter to start another new game`;
+    return myOutputValue;
+  }
+};
+
+//Beat That Game Function
+var beatThatGame = function () {};
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  var myOutputValue = "";
+
+  if (currentGameMode == "welcomeToBeatThat") {
+    //Opening message + Instructions
+    myOutputValue = `Welcome to BEAT THAT!<br><br>
+     BEFORE starting the game, please ENTER desired player 1 In Game Name (IGN) and click ENTER, followed by player 2. <br><br>
+      Game Rules: <br>
+      1. This is a 2 person, player vs player game. <br>
+2. The game will roll 2 dice and show the dice rolls result. For example, dice 1 rolled 3 and dice 2 rolled 4. <br>
+3. The player that rolled the dice will pick the order of the dice.(If they wanted the number 43, they would specify that the 2nd dice goes first.) <br>
+4. After both players have rolled and chosen their dice order, the player with the higher combined number wins.  `;
+
+    currentGameMode = "enterPlayer1";
+
+    return myOutputValue;
+  }
+  if (currentGameMode == "enterPlayer1") {
+    //Set player 1 IGN
+    player1 = input;
+
+    currentGameMode = "enterPlayer2";
+
+    console.log(`currentGameMode, ${currentGameMode}`);
+    console.log(`Player 1 IGN, player 1: ${player1}`);
+
+    myOutputValue = `Welcome ${player1}. Player 2, please enter your desired IGN and click enter. `;
+  } else if (currentGameMode == "enterPlayer2") {
+    //Set player 2 IGN
+    player2 = input;
+
+    currentGameMode = "player1Roll"; //Switch to Game Result for player 1
+
+    console.log(`currentGameMode, ${currentGameMode}`);
+    console.log(`Player 2 IGN, player 2: ${player2}`);
+
+    myOutputValue = `Welcome ${player2}. <br><br>
+    ${player1}, you may begin to start the game by clicking enter.`;
+  } else if (
+    // Condition for player 1 turn
+    //player 1 turn + player 1 result
+    currentGameMode == "player1Roll" || //Dice results of player 1
+    currentGameMode == "player1Choice" //player 1 desired outcome
+  ) {
+    userChosenDiceOrder = input;
+    myOutputValue = player1Result(player1, player2, userChosenDiceOrder);
+    return myOutputValue;
+  } else if (
+    //player 2 turn + player 2 result
+    currentGameMode == "player2Roll" || //Dice results of player 2
+    currentGameMode == "player2Choice" //player 2 desired outcome
+  ) {
+    userChosenDiceOrder = input;
+    myOutputValue = player2Result(player2, userChosenDiceOrder);
+    return myOutputValue;
```

Comment:
similar to above, a bit repetitive, consider how to refactor this into just a single `else if` block